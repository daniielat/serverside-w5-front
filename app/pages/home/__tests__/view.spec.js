const React = require('react');
const HomeView = require('../view');
const { render, screen } = require('@testing-library/react');
const mockProducts = require('../../../../utils/mockProductsEI');

describe('Ejercicio 4 - La View de /home', () => {
    let component;
    const i18n = { gettext: text => text };
    beforeEach(() => {
        component = render(<HomeView i18n={i18n} title='Productos' products={[]}/>)
    })
    it('1) Renderiza', () => {
        const { asFragment } = component;
        expect(asFragment()).toMatchSnapshot();        
    })

    it('2) Muestra en pantalla un <h2> con la prop `title`', () => {
        const title = screen.getByText('Productos');
        expect(title).toBeInTheDocument();
    })
})

describe('EjericitaciÃ³n Integradora - La View de /home', () => {
    let component;
    const i18n = { gettext: text => text };
    beforeEach(() => {
        component = render(<HomeView i18n={i18n} title='Productos' products={mockProducts}/>)
    });

    it('1) Renderiza', () => {
        const { asFragment } = component;
        expect(asFragment()).toMatchSnapshot();        
    });

    it('2) Muestra un array de productos', () => {
        const products = screen.getAllByRole('listitem');
        expect(products.length).toBe(10);
    })
})