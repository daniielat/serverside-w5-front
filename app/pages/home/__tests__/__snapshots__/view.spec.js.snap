// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ejercicio 4 - La View de /home 1) Renderiza 1`] = `
<DocumentFragment>
  <div
    class="demo"
  >
    <h2>
      Productos
    </h2>
    <ul />
  </div>
</DocumentFragment>
`;

exports[`Ejericitaci√≥n Integradora - La View de /home 1) Renderiza 1`] = `
<DocumentFragment>
  <div
    class="demo"
  >
    <h2>
      Productos
    </h2>
    <ul>
      <li>
        Motorola
      </li>
      <li>
        Samsung
      </li>
      <li>
        Huawei
      </li>
      <li>
        iPhone
      </li>
      <li>
        Samsung2
      </li>
      <li>
        Nokia
      </li>
      <li>
        LG
      </li>
      <li>
        Sony
      </li>
      <li>
        Motorola2
      </li>
      <li>
        Moto-G
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
